test/conformance/kuttl/generate/clusterpolicy/cornercases/clone-list-sync-same-trigger-source-update-source/03-update-trigger.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    location: europe
+    allowedToBeCloned: "true"
+  annotations:
+    myProj/cluster.addon.sync.targetNamespace: no-clone
```

Comment:
Assuming there was a Namespace named `no-clone`, after this update is made wouldn't we expect to see two things?

1. resource is deleted from `clone-list-sync-same-trigger-source-update-source-target-ns` Namespace
2. resource is generated into `no-clone` Namespace

If this is the case, maybe it also makes sense to incorporate that second check here?

Reply:
Good point! I changed the value (not remove it) because the kuttl apply does not remove the annotation when it is removed from the manifest. I can add assertion for 2.

Reply:
Actually no, as the rule does not match the namespace `no-clone`, so there shouldn't be any new secret created.

Reply:
Oh right, I didn't catch that. Do you think it's worth modifying this (and the regular generate-clone) test so that this behavior could be proven? It would entail modifying the policy in the first step to just match on `"?*"` for the annotation's value.

Reply:
If we change the rule to match the annotation value `"?*"`, the old target will not be cleaned up as it still matches the trigger, but a new target will be generated. Would you like to update the test to cover that?

Reply:
Right, of course. Let's go with this.

Reply:
Ok, I will push the change. Also saw another flaky test, will fix it.

Reply:
Done and done.